meta-client/src/main/java/org/apache/helix/metaclient/api/MetaClientCacheInterface.java

```diff
@@ -27,19 +27,29 @@ public interface MetaClientCacheInterface<T> extends MetaClientInterface<T> {
     /**
      * TrieNode class to store the children of the entries to be cached.
      */
-    class TrieNode {
+    public class TrieNode {
         // A mapping between trie key and children nodes.
         private Map<String, TrieNode> _children;
 
         // the complete path/prefix leading to the current node.
         private final String _path;
 
         private final String _nodeKey;
+        private boolean _childrenCached;
 
-        TrieNode(String path, String nodeKey) {
+        public TrieNode(String path, String nodeKey) {
             _path = path;
             _nodeKey = nodeKey;
             _children = new HashMap<>();
+            _childrenCached = false;
+        }
+
+        public boolean isChildrenCached() {
```

Comment:
Consider AtomicBoolean

Reply:
Removal of AtomicBoolean as it is no longer needed when populating all cache from the beginning